raiden/messages.py

```diff
@@ -1662,6 +1662,37 @@ def unpack(
         )
         return monitoring_request
 
+    def verify(self, partner_address, requesting_address):
+        balance_proof_data = pack_balance_proof(
```

Comment:
Is this function supposed to be called by the monitoring service or the client?

Reply:
Any party that wants to verify signatures/integrity of the message.

Reply:
This is how the MS would check a received message, right?

Reply:
> Any party that wants to verify signatures/integrity of the message.

> This is how the MS would check a received message, right?

The `MonitoringService` should apply additional checks in order to avoid sending onchain transactions that get rejected, for instance also for the case where to nodes sign a `balance_proof` that doesn't work onchain. 

The idea of the test here is a little different though, i.e. that if the `MonitoringService` calls this function and the verification fails, it can not know if it failed because the client sent an invalid message or the message was tempered with on the way. That's in my understanding one purpose of signatures/authentication. The MS can determine that, if `non_closing_signature`, `balance_hash` and `additional_hash` are added to` RequestMonitoring._data_to_sign`

Reply:
> should apply additional checks in order to avoid sending onchain transactions that get rejected

How can you determine that a BP will get rejected on-chain?

Reply:
> How can you determine that a BP will get rejected on-chain?

A hard way would be to execute a transaction locally before sending it. 
An easier way would be to redo the checks that the client/contract does, e.g. checking for overflows, verifying signatures, verify that the `balance_hash` was computed correctly..  

If I were an RSB provider and every failed transaction would make me loose some cents and potentially the reward, I'd like to have these checks in place.

Reply:
> A hard way would be to execute a transaction locally before sending it.
> How can you determine that a BP will get rejected on-chain?

You can't really be 100% sure a transaction won't get rejected on-chain. Too many race conditions exist that can happen. Simplest one would be another MS or the user themselves beating you to it.